cnd/src/btsieve/ethereum/mod.rs

```diff
@@ -24,16 +20,148 @@ use std::collections::HashSet;
 type Hash = H256;
 type Block = crate::ethereum::Block<Transaction>;
 
-pub async fn matching_transaction<C>(
+pub async fn matching_create_contract<C>(
```

Comment:
:thinking: Probably for a follow-up PR, but what do you think about renaming this function to `look_for_contract_creation` and doing something similar for the other ones? I find the term matching *so* confusing because it's both and adjective and a verb.

Reply:
I am onboard with this! Alternative ways could also be:

- `wait_for_contract_creation`
- `watch_for_contract_created`

Reply:
Well spotted @lucas, choosing 'watching' as a nice verb for what we are doing.

Reply:
Renamed:
- `matching_create_contract` -> `watch_for_contract_creation`
- `matching_event` -> `watch_for_event`

But that leaves the functions
- `matching_transaction_and_log`
- `matching_transaction_and_receipt`

Also the `matcher` predicate function will need to be renamed to 'match' the functions (pun intended :)

Any ideas?  Otherwise I'll have another think tomorrow and submit follow up PR.

Reply:
I vote for follow-up PR!

Let's get this in! :tada: